core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/util/ValueComparator.java

```diff
@@ -69,8 +70,21 @@ public int compare(Value o1, Value o2) {
 			return 1;
 		}
 
-		// 4. RDF literals
-		return compareLiterals((Literal) o1, (Literal) o2);
+		// 4. Literals
```

Comment:
@jeenbroekstra why do we have two ValueComapartors?

Reply:
> @jeenbroekstra why do we have two ValueComapartors?

I'm probably overlooking something. What other one besides this class do you see?

Reply:
Never mind, my fault. I had mistaken the LexicalValueComparator for being SPARQL compliant.